### ISSUE TITLE
How can we customize Preview (content) using master layout

### ISSUE THREAD

**Issue by webmedia1012** [OrchardCMS/OrchardCore#8115]
Hi Guys,

Trying to find a documentation on this, the content preview (before you publish) works, the only thing is that it's just on a page without any css or our site layout (headers and footers), how to I make it so I can see what it's going to look like on the actual site?

I saw that it was rendering on orchardcore.contentpreview/preview/index/id - can I overload or extend this page and put it on my views folder? so I can view it with our own layout?

Thank you,
Gil

**Comment by jtkech** [comment]
Normally in that case, you need to add a preview part to your content type, and then in the settings specify the preview url / path

**Comment by webmedia1012** [comment]
thank you @jean can you elaborate on that? when you say content type, create a new content type from content definition? I don't follow.

**Comment by ns8482e** [comment]
@webmedia1012 `Preview` uses the front-end theme.  As @jtkech suggested - If you created site using **Decoupled** approach using Blank site then you need to add `PreviewPart` to your content type.

 https://docs.orchardcore.net/en/dev/docs/guides/decoupled-cms/#setting-up-a-new-site

**Comment by webmedia1012** [comment]
Ok I think I know where this is but, when I go to Add Part, all the parts (alias etc) are there except for Preview, then I go check the Configuration ->features the Preview is enabled, am I missing something else? Thanks guys !

**Comment by jtkech** [comment]
Hmm, be sure that the Preview feature is enabled, in the Features page if you see an `Enabled` button, it doesn't mean that the feature is enabled but that you need to click on it to enable it

**Comment by webmedia1012** [comment]
I see disabled button , that means it's enabled correct? It's so weird, also just to test, I setup a quick project and created a new site, selecting CMS headless (same setup with our website) and I can see the preview part, but with the current site I cannot. Is there a package I'm missing maybe?

**Comment by jtkech** [comment]
Can you see this part in the page that lists all existing parts, content definition => content parts, if so edit it to see if it is still marked as `Attachable`

**Comment by webmedia1012** [comment]
did you say content definition? that's the one on the app_data/default correct? would it make any difference if I disabled it and move it to the database? but let me look.

**Comment by webmedia1012** [comment]
Ok nevermind it's not even on the content definition. unless theres content definition inside of the admin

**Comment by jtkech** [comment]
@webmedia1012 

I was talking about this admin page

![Preview1](https://user-images.githubusercontent.com/8586360/118283868-1b59cf80-b4d0-11eb-8858-9368c3d9c27c.png)

And then this setting

![Preview2](https://user-images.githubusercontent.com/8586360/118283890-21e84700-b4d0-11eb-9430-216a0760b089.png)

**Comment by webmedia1012** [comment]
thank you so much @jtkech here's my complete part list.
![capture_cms](https://user-images.githubusercontent.com/2730217/118311433-bfc61c80-b4a4-11eb-9d34-640271f04ffb.PNG)

**Comment by webmedia1012** [comment]
.. and here's my preview on features. 

![image](https://user-images.githubusercontent.com/2730217/118311561-eb490700-b4a4-11eb-92fd-db8e43d47625.png)

**Comment by jtkech** [comment]
Oh yes, now i remember, when this feature was done the preview part was not there, it has been added afterwards but only on the dev branch, so i assume you use the packages from the Nuget feed, you would need to use the dev packages from the cloudsmith feed, or wait for our incoming version 1.0 to be published on the Nuget feed.

**Comment by webmedia1012** [comment]
Ah ok that makes sense now, yes I use the nuget package feed, how do I switch to dev packages is it going to mess up some of the dependencies? Thanks so much @jtkech

**Comment by webmedia1012** [comment]
I upgraded and this is what I got, how to I correct this? 
![image](https://user-images.githubusercontent.com/2730217/118316252-7200e280-b4ab-11eb-9a6c-024e5d5d51ba.png)

**Comment by jtkech** [comment]
See in our repo under the root folder, there is a `Nuget.config` file with a commented out line

    

Under the root of your solution you need a `Nuget.config` file specifying the cloudsmith feed

    <add key="OrchardCore" value="https://nuget.cloudsmith.io/orchardcore/preview/v3/index.json" />

**Comment by webmedia1012** [comment]
Oh ok easy enough lol.. thanks let me try and report back if this is successful will close this out.

**Comment by jtkech** [comment]
Yes, you can use the last version, some migrations will run so do a backup before, your app_data folder / database.

**Comment by webmedia1012** [comment]
Hi @jtkech  thanks again. After I changed the nuget.config on my root to include this..
 <add key="OrchardCore" value="https://nuget.cloudsmith.io/orchardcore/preview/v3/index.json" />

and went to Manage nuget package on my project, I saw the rc-16202 version but when I installed it.. I get this error,
do I need to 1. remove the dlls from the old version? 2. I've removed old version from project settings, do I need to add this one? 

Thanks! I feel like I'm getting close lol

![image](https://user-images.githubusercontent.com/2730217/118544007-e46efe00-b709-11eb-94b3-5a1768457bb5.png)
